fe/fe-core/src/main/java/com/starrocks/catalog/IcebergTable.java

```diff
@@ -170,6 +170,15 @@ public List<Column> getPartitionColumnsIncludeTransformed() {
         return allPartitionColumns;
     }
 
+    public boolean isAllPartitionColumnsIdentity() {
+        for (PartitionField field : getNativeTable().spec().fields()) {
```

Comment:
getNativeTable just get a current snapshot, partition column is identity in this snapshot not means that it used to be identity.